assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/TotalUserCount.js

```diff
@@ -83,7 +92,8 @@ export default function TotalUserCount( { loaded, error, report, dimensionValue
 				{ ! dimensionValue && __( 'All Users', 'google-site-kit' ) }
 				{ dimensionValue && (
 					<Fragment>
-						{ __( 'All Users', 'google-site-kit' ) }
+						{ dimensionValue && <button onClick={ showAllUsers }>{ __( 'All Users', 'google-site-kit' ) }</button> }
```

Comment:
This should use the `Link` component instead - in general, we should always rely on `Link` or `Button`, never on the plain HTML elements. Both of our components work correctly regardless of whether semantically we want a link (`a`) or a button (`a`) - the component name is based on the appearance (e.g. a `Link` always looks like a link should look, no matter if it technically only has no `href` and would therefore use the `button` HTML element under the hood).

By using this component, we should also get the correct cursor (referring to your comment).